charts/golang/templates/deployment.yaml

```diff
@@ -21,8 +21,8 @@ spec:
   {{- end }}
   template:
     metadata:
-      {{- if or .Values.podAnnotations .Values.vault.enabled }}
       annotations:
+        rollme: {{ randAlphaNum 5 | quote }}
```

Comment:
This is the [official way](https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments).

Reply:
I get it, but it still feels kludgy somehow :(

Reply:
@GrantSheehan If they did not deprecate the `--recreate-pods` in Helm v3, we could detect a rekick and provide that flag to avoid this... but thats life. 😢